sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateFormatter.scala

```diff
@@ -33,6 +34,8 @@ sealed trait DateFormatter extends Serializable {
   def format(days: Int): String
   def format(date: Date): String
   def format(localDate: LocalDate): String
+
+  def initialize(): Unit = {}
```

Comment:
shall we force all the children to implement `initialize`?

And maybe a better name is `validatePatternString`. `initialize` sounds like it must be called.

Reply:
and we should call it in `TimestampFormatter.apply` if the policy is not legacy, to fail earlier.

Reply:
Got it. SGTM.